src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_field_from_props.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { useMemo, useEffect } from 'react';
+import { UseFieldProps } from '../components';
+import { FieldHook, FieldConfig, FieldValidationData } from '../types';
+import { useFormContext } from '../form_context';
+import { useField, InternalFieldConfig } from './use_field';
+
+/**
+ * Hook to initialize a FieldHook based on Props passed to <UseField /> or <UseMultiFields />
+ *
+ * @param props The props passed to <UseField /> or <UseMultiFields />
+ * @returns The field hook and props to forward to component to render for the field
+ */
+
+export const useFieldFromProps = <T, FormType, I>(
```

Comment:
Initially the code in this hook was inside the <UseField /> component.

It serves to initialise a FieldHook (`useField()`) using the props passed to the component.

When refactoring the <UseMultiFields /> component I realised that it would be much simpler to extract the code from `<UseField />` into a separate hook and consume it from both places.